src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2216,7 +2216,7 @@ public static void Select(IList checkRead, IList checkWrite, IList checkError, i
             // Validate input parameters.
             if ((checkRead == null || checkRead.Count == 0) && (checkWrite == null || checkWrite.Count == 0) && (checkError == null || checkError.Count == 0))
             {
-                throw new ArgumentNullException(SR.net_sockets_empty_select);
+                throw new ArgumentNullException(null, SR.net_sockets_empty_select);
```

Comment:
Seems like this one would be better suited by being split up into multiple if checks/blocks (it's also odd that an ArgumentNullException is being used for the Count == 0 case), e.g.
```C#
if (checkRead == null || checkRead.Count == 0)
{
    throw new ArgumentNullException(nameof(checkRead), SR.net_sockets_empty_select);
}
if (checkWrite == null || checkWrite.Count == 0)
{
    ... // etc.
```
@cipop? @davidsh?